### ISSUE TITLE
Rclone 1.65 fails to list the contents of a specific folder which contains an onenote file

### ISSUE THREAD

**Issue by pitsi** [rclone/rclone#7499]
#### The associated forum post URL from `https://forum.rclone.org`
Not a member of the forum.

#### What is the problem you are having with rclone?
Rclone 1.65 fails to list the files of a specific folder, as you can see in the log below. The folder does contain an onenote file, but rclone had no issues listing its contents with previous versions. It also contains docx, pdf, jpg, png and similar files. To be exact, "Έγγραφα" means "Documents" in greek, so it contains all sorts of documents I keep close at hand.

I discovered the issue earlier, when I wanted to download 2 documents from there via rclone-browser. Clicking the folder returned no files under it. That is why I had to use the site to download my files.

Last but not least, since debian's package for rclone is still at 1.60.1, I use the deb package from here and I have to manually update it. I do not remember if I was on the latest 1.64.x version, but I definitely was on 1.64 last time I accessed that folder.

#### What is your rclone version (output from `rclone version`)
```
$ rclone version
rclone v1.65.0
- os/version: debian trixie/sid (64 bit)
- os/kernel: 6.5.0-5-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.4
- go/linking: static
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Debian testing/unstable x64.

#### Which cloud storage system are you using? (e.g. Google Drive)
Onedrive.

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
```
$ rclone ls onedrive:Έγγραφα
2023/12/10 15:28:46 ERROR : : error listing: unknown object type <nil>
2023/12/10 15:28:46 Failed to ls with 2 errors: last error was: unknown object type <nil>
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
$ rclone ls onedrive:Έγγραφα -vv
2023/12/10 15:29:28 DEBUG : rclone: Version "v1.65.0" starting with parameters ["rclone" "ls" "onedrive:Έγγραφα" "-vv"]
2023/12/10 15:29:28 DEBUG : Creating backend with remote "onedrive:Έγγραφα"
2023/12/10 15:29:28 DEBUG : Using config file from "/home/user/.config/rclone/rclone.conf"
2023/12/10 15:29:29 DEBUG : Σημειωματάριο του χρήστη user: OneNote file not shown in directory listing
2023/12/10 15:29:29 ERROR : : error listing: unknown object type <nil>
2023/12/10 15:29:29 DEBUG : 7 go routines active
2023/12/10 15:29:29 Failed to ls with 2 errors: last error was: unknown object type <nil>
```

**Comment by ncw** [comment]
This is fixed in the beta and will be released in the 1.65.1 release which should be quite soon, just waiting for one more fix

**Comment by Animosity022** [comment]
Already fixed and reported.

**Comment by pitsi** [comment]
Is this the issue report you refer to?
#7456 

Because I checked the character with the python command mentioned in the forum thread and this is what it returns
```
>>> ord("'")
39
```
It is the greek hyphenation... mark in my case, but it still counts as a special character.

**Comment by tigerjack** [comment]
Hi @Animosity022 can you point out the issue you are referring to?
Additionally, is there a way to download a patch while waiting for the new version to be distributed?

**Comment by pitsi** [comment]
Hello and Merry Christmas to everyone!

I downloaded the latest beta from beta.rclone.org, the one that was built last week, and I can verify that it works
```
$ ./rclone --version
rclone v1.66.0-beta.7580.8503282a5
- os/version: debian trixie/sid (64 bit)
- os/kernel: 6.5.0-5-amd64 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.21.5
- go/linking: static
- go/tags: none

$ ./rclone ls onedrive:Έγγραφα -vv
2023/12/25 12:51:03 DEBUG : rclone: Version "v1.66.0-beta.7580.8503282a5" starting with parameters ["./rclone" "ls" "onedrive:Έγγραφα" "-vv"]
2023/12/25 12:51:03 DEBUG : Creating backend with remote "onedrive:Έγγραφα"
2023/12/25 12:51:03 DEBUG : Using config file from "/home/user/.config/rclone/rclone.conf"
2023/12/25 12:51:03 DEBUG : Σημειωματάριο του χρήστη Dimitris: OneNote file not shown in directory listing
(loads of pdf, docx, xlsx etc files here)
2023/12/25 12:51:03 DEBUG : 7 go routines active
```

**Comment by pitsi** [comment]
As it seems, the commit that fixed the onedrive issue was this one
https://github.com/rclone/rclone/commit/f0c774156e5f4d4f51492756c7b1ebf2bc8b98d0

The build that contains this commit, i.e. v1.66.0-beta.7560.**f0c774156**, is probably the oldest one that fixes the issue. I just tested it and it works the same as above.
My question now is: was it really the onenote file that caused the issue or the character I mentioned above?

**Comment by pitsi** [comment]
Yesterdays upgrade to v1.65.1 fixed the issue for the stable version of rclone.

**Comment by ncw** [comment]
@pitsi yes this was released in v1.65.1 - sorry for the delay - the point release was tricker than usual!
